src/common/telemetry/collect-telemetry/diff.ts

```diff
@@ -0,0 +1,63 @@
+type ObjectLike = Record<string, unknown>
+
+const isObject = (o: unknown): o is ObjectLike => o != null && typeof o === 'object'
+
+/**
+ * Compares two objects and returns the difference.
+ * - new and updated properties look similar
+ *  (they are defined on the difference object with the final value);
+ * - deleted properties are shown as null.
+ */
+export const diff = (initial: ObjectLike, updated: ObjectLike): ObjectLike => {
```

Comment:
@quesabe aren't there ready-to-use libs available?

Reply:
There are, however I found most of them lacking in some functionality and clarity of output format. Thus I decided it's just easier to create a solution, that suits us, instead of looking for other libraries (that would probably need some tuning any way).